src/test/java/com/bobocode/hoverla/bring/context/BeanValidatorTest.java

```diff
@@ -0,0 +1,160 @@
+package com.bobocode.hoverla.bring.context;
+
+import com.bobocode.hoverla.bring.exception.BeanValidationException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class BeanValidatorTest {
+
+    private static final String BD1 = "beanDef1";
+    private static final String BD2 = "beanDef2";
+    private static final String BD3 = "beanDef3";
+    private static final String BD4 = "beanDef4";
+    private List<BeanDefinition> beanDefinitionList;
+    private BeanValidator beanValidator;
+
+    @BeforeEach
+    public void setUp() {
+        beanValidator = new BeanValidator();
+        BeanDefinition beanDef4 = prepareDefinition(BD4, Integer.class);
+        BeanDefinition beanDef3 = prepareDefinition(BD3, Byte.class, beanDef4);
+        BeanDefinition beanDef2 = prepareDefinition(BD2, Short.class, beanDef3, beanDef4);
+        BeanDefinition beanDef1 = prepareDefinition(BD1, String.class, beanDef2, beanDef3);
+        beanDefinitionList = new ArrayList<>(Arrays.asList(beanDef1, beanDef2, beanDef3, beanDef4));
```

Comment:
(nit) List.of() instead of new ArrayList<>(Arrays.asList())

Reply:
done